### DISCUSSION TITLE
CLOSED: ImageMagick-Windows10- no decode delegate for this image format 'JPEG'

### DISCUSSION THREAD

**Discussion by Paxel-photo** [ImageMagick/ImageMagick#7019] [Help] [answered]
### ImageMagick version

7.1.1-25 Q16 x64

### Operating system, version and so on

Édition Windows 10 Professionnel Version	22H2

### Description

# No delegates for image format JPEG
## Windows 10 professionnal
Hi all,
Freshly installed on my machine ImageMagick returns an error "no decode delegate for this image format 'JPEG' @ error/constitute.c/Readlmage/746".
I tried from a Lightroom Plugin (LR Mogrify) but I also tried IMDisplay and the commandline version for magick.exe. All returned the same error.

I have searched for similar discussions, but they are related to Linux OS, with extra needs for installation, that I don't think are required in Windows, as magick itself reports the delegates as "built-in".

The OS is Windows 10 professionnal.
<code>Édition Windows 10 Professionnel
Version	22H2
Installé le	‎24/‎01/‎2023
Build du système d’exploitation	19045.3803
Expérience Windows Feature Experience Pack 1000.19053.1000.0
</code>

Result of "magick identify -version" command:
<code>Version: ImageMagick 7.1.1-25 Q16 x64 98e7513:20231230 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Channel-masks(64-bit) Cipher DPC Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png ps raqm raw rsvg tiff webp xml zip zlib
Compiler: Visual Studio 2022 (193732826)</code>

Note: I tried with several JPEG images, but also with PNG images, the result is the same.

Here's the last attempt:
<code>C:\Program Files\ImageMagick-7.1.1-Q16>magick test\LR.jpg -resize 100x100 test\LR_logo.png
magick: no decode delegate for this image format \`JPEG\' @ error/constitute.c/ReadImage/746.</code>

So the delegates are supposed to be built-ins ones. 
How can I get such an error? 
Any idea of a solution?

**Comment by dlemstra** [comment]
I suspect there is an issue with your installation of ImageMagick. You might want to uninstall it and try again? It's also possible that something else set environment variables that affect your installation.

**Reply by Paxel-photo** [reply]
Thanks for the idea.
I uninstalled ImageMagick, restarted the PC, then I did a new install with another version (HDR) of ImageMagick.
<code>
Version: ImageMagick 7.1.1-25 Q16-HDRI x64 98e7513:20231230 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Channel-masks(64-bit) Cipher DPC HDRI Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png ps raqm raw rsvg tiff webp xml zip zlib
Compiler: Visual Studio 2022 (193732826)
</code>
But unfortunately:
<code>C:\Program Files\ImageMagick-7.1.1-Q16-HDRI>magick test\LR.jpg test\LR.png
magick: no decode delegate for this image format `JPEG' @ error/constitute.c/ReadImage/746.</code>

So I had a look at the PATH env var:
<code>PATH=C:\Program Files\ImageMagick-7.1.1-Q16-HDRI;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;C:\Program Files\Intel\WiFi\bin\;C:\Program Files\Common Files\Intel\WirelessCommon\;C:\WINDOWS\System32\OpenSSH\;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\Program Files\Calibre2\;C:\Program Files\Gohstscript\gs10.00.0\bin;C:\Program Files\PuTTY\;C:\Program Files (x86)\X-Rite\Devices\Services;C:\Program Files (x86)\X-Rite\Devices\Lib;C:\Users\patrick\AppData\Local\Microsoft\WindowsApps;</code>
and I found no clue.

I suspected a problem of authorisation; so I ran the program through runas, with an admin user account:
<code>runas /user:patadmin "magick 'test\LR.jpg test\lr.png'"
Entrez le mot de passe de patadmin :
Tentative de lancement de magick 'test\LR.jpg test\lr.png' en tant qu’utilisateur "PCNAME\patadmin" ...</code>
That time I just see a flash on my screen: a window was opened and closed immediately; no mean to see what it displayed.

So I tried : 
<code>runas /user:patadmin "imdisplay"
Entrez le mot de passe de patadmin :
Tentative de lancement de imdisplay en tant qu’utilisateur "PCNAME\patadmin" ...</code>
imDisplay opens, I can select a file 
![image](https://github.com/ImageMagick/ImageMagick/assets/26047529/8b383edf-5c69-4a89-8147-3809e5f1a369)
but when I try to open it:
![image](https://github.com/ImageMagick/ImageMagick/assets/26047529/64aeaf1c-4d70-4a7d-99c7-862d1d65ca7c)

So the problem is still there.

**Reply by dlemstra** [reply, marked as answer]
You might also want to take a look in your registry or run `magick -debug all` to get more verbose output.

**Reply by Paxel-photo** [reply]
Could you tell what I could search in the registry?
Have you an idea or it just a feeling?
Any case, thanks for your time and attention, it's good to feel some support.

**Reply by dlemstra** [reply]
I think using `-debug all` on the command line should give you some information on where it tries to load the modules for `jpeg`. And you should then search your registry for that path.

**Reply by Paxel-photo** [reply]
Finally I got it, but it seems there is an issue to open !

Thanks to your advice to use the debug option, I was warned by the fact it is searching the DLL a lot of times.
I have tried to add the "coders" directory to the system path -> no result
I have tried to move the modules directly to to the "IMagick" directory -> no result

So I uninstalled it once more, and this time I installed a "static" version, without DLLs. -> **It works fine.**
So my own problem is solved, but that leaves unchanged the fact that I had no way to have a DLL version working on my config.

Many thanks.

**Reply by dlemstra** [reply]
This is the first time that you mention `IMagick` so I suspect that this is related to your setup. Please don't open an issue. It should work on your machine but there is something on your machine that prevents it from working.

**Reply by Paxel-photo** [reply]
Hello, no, "IMagick" was just a short writing for "C:\Program Files\ImageMagick-7.1.1-Q16-HDRI-DLL".
I really think there is an issue, but I have no way to describe it, so if you can do it, then go.
Thanks for your help @dlemstra!

**Reply by dlemstra** [reply]
The issue is related to your system. There is something on/in your system that breaks ImageMagick. If this would not work on other machines we would have gotten a lot more issues.

**Reply by Yanncha** [reply]
Hello,

For information, I just came into a kinda same issue on Windows 7 professional, with the ImageMagick-7.1.1-29-Q16-HDRI-x64-dll version.
Though in my case, it was working fine when manually calling magick from the CMD box, but failing when doing so through a VBA script (with error "no decode delegate..." if I got the error output back).

The problem was solved, as above, by installing rather the static version ImageMagick-7.1.1-29-Q16-x64-static.

Best regards

**Reply by vpalichev** [reply]
Had the same issue, Windows 10, installed the version recommended on the official download page -- [ImageMagick-7.1.1-43-Q16-HDRI-x64-dll.exe](https://imagemagick.org/archive/binaries/ImageMagick-7.1.1-43-Q16-HDRI-x64-dll.exe) and the same problem, no image format can be processed, all the same "no decode delegate..." every time, even with BMP format!
In my case static version of ImageMagick did not help, still looking for solution.

**Reply by fmw42** [reply]
Show what you get from

`magick -version`

**Reply by vpalichev** [reply]
Hello, Fred, thanks for swift reply.
My story got an interesting and unexpected conclusion:

After some trial and frustration I figured that this command runs silently and as intended:
`magick   file.png  -fill white -opaque blue   file_new.png`

While this command  with HEX color specified, which I pretty much just copied from some forum:
`magick rubiks.png  -fill '#F9E231' -opaque '#60D838'   rubics_new.png`

will give you a, well, slightly misleading error:
``magick: no decode delegate for this image format `' @ error/constitute.c/ReadImage/746.``

After replacing **single quotes** with **double quotes** things worked fine on Windows.
Maybe it's a trivial knowledge that Windows and Linux have different quoting format, but the error message took me by surprise.

Here's a quote from documentation:
> However, in most cases the roles of single-quotes and double-quotes are reversed with respect to Linux and Windows

[Further obligatory reading](https://imagemagick.org/script/command-line-processing.php)
